### PR TITLE
Tensorflow r1.4

### PR DESCRIPTION
**Pull request by lissyx**

### CODE REVIEW

File: tc-tests-utils.sh [hunk 1]
```diff
@@ -206,9 +211,13 @@ do_get_model_parameters()
 
   wget "${model_url}" -O "${model_file}"
   wget "${SUMMARIZE_GRAPH_BINARY}" -O "/tmp/summarize_graph"
+  wget "$(dirname ${SUMMARIZE_GRAPH_BINARY})/libtensorflow_framework.so" -O "/tmp/libtensorflow_framework.so"
```
Comment (reuben) [inline tc-tests-utils.sh]: Please add a separate variable for it.

**Comment by reuben** [comment]
Hm. With this PR we still can't use the decoder library with upstream TensorFlow, but this time because we build with CUDA 9 and they build with CUDA 8.

**Comment by reuben** [comment]
Ah, putting both versions in LD_LIBRARY_PATH works. I'm happy.

File: native_client/README.md [hunk 2]
```diff
@@ -72,7 +72,7 @@ Before building the DeepSpeech client libraries, you will need to prepare your e
 Then you can build the Tensorflow and DeepSpeech libraries.
 
 ```
-bazel build -c opt --copt=-O3 //tensorflow:libtensorflow_cc.so //native_client:deepspeech //native_client:deepspeech_utils //native_client:ctc_decoder_with_kenlm //native_client:generate_trie
+bazel build -c opt --copt=-O3 //tensorflow:libtensorflow_cc.so //tensorflow:libtensorflow_framework.so //native_client:deepspeech //native_client:deepspeech_utils //native_client:ctc_decoder_with_kenlm //native_client:generate_trie
```
[the remark this answers is not in the document]
Comment (lissyx) [reply native_client/README.md]: Good catch

File: native_client/model_size.sh [hunk 3]
```diff
@@ -5,7 +5,7 @@
 #  $2 => dimension to use for timesteps
 #  $3 => dimension to use for framesize
 if [ $# -ne 3 ]; then
-    exit 1
+    exit 42
```
[the remark this answers is not in the document]
Comment (lissyx) [reply native_client/model_size.sh]: Leftover from debugging

File: tc-tests-utils.sh (hunk 1 above)
Comment (lissyx) [reply tc-tests-utils.sh]: Addressing that.

File: native_client/BUILD [hunk 4]
```diff
@@ -109,9 +109,11 @@ cc_library(
     includes = ["kenlm", "boost_locale"],
     copts = ["-std=c++11"],
     defines = ["KENLM_MAX_ORDER=6"],
-    deps = ["//tensorflow/core:core",
+    deps = ["//tensorflow/core:framework_headers_lib",
             "//tensorflow/core/util/ctc",
             "//third_party/eigen3",
+	    "@protobuf_archive//:protobuf_headers",
+	    "@nsync//:nsync_headers",
```
[the remark this answers is not in the document]
Comment (lissyx) [reply native_client/BUILD]: Done.

**Comment by lissyx** [comment]
@reuben All green and addressed your comments :)

**Pull request merged**